src/drivers/driver_zephyr.c

```diff
@@ -122,15 +122,25 @@ void wpa_drv_zep_event_proc_scan_res(struct zep_drv_if_ctx *if_ctx,
 	struct wpa_scan_res **tmp = NULL;
 	size_t scan_res_len  = sizeof(struct wpa_scan_res) + r->ie_len + r->beacon_ie_len;
 
-	if (!if_ctx->scan_res2)
-		return;
+	if (!if_ctx->scan_res2) {
+		if_ctx->scan_res2 = os_zalloc(sizeof(*if_ctx->scan_res2));
+		wpa_printf(MSG_INFO, "%s: Alloc memory for scan results\n", __func__);
+	}
+
+
+	if (!if_ctx->scan_res2) {
```

Comment:
Can we move this check immediately after the os_zalloc?

Reply:
This was intentional, if it's either non-NULL or allocation failure we need to catch that.